tests/network/test_reaction_network.py

```diff
@@ -823,38 +824,120 @@ def test_mols_w_cuttoff(self):
-        ) as handle:
-            mols_to_keep = pickle.load(handle)
+        for node in nodes:
+            r_p = ReactionNetwork.parse_reaction_node(node)
+            node_prod_react.append(r_p)
 
-        reactions = ReactionNetwork.identify_concerted_rxns_via_intermediates(
-            RN_loaded, mols_to_keep, single_elem_interm_ignore=["C1", "H1", "O1", "Li1"]
+        self.assertListEqual(
+            node_prod_react,
+            [([19, 32], [673]), ([41], [992]), ([1, 652], [40, 53]), ([4], [5, 6])],
         )
 
-        self.assertEqual(len(reactions), 2410)
+    def test_generate_node_string(self):
 
-    def test_add_concerted_rxns(self):
-        with open(os.path.join(test_dir, "unittest_RN_pr_solved.pkl"), "rb") as input:
-            RN_pr_solved = pickle.load(input)
+        react_prod = [
+            ([19, 32], [673]),
+            ([41], [992]),
+            ([1, 652], [40, 53]),
+            ([4], [5, 6]),
+        ]
+        node_strings = []
 
-        RN_loaded = copy.deepcopy(RN_pr_solved)
+        for rxn in react_prod:
+            node_str = ReactionNetwork.generate_node_string(rxn[0], rxn[1])
+            node_strings.append(node_str)
+
+        self.assertListEqual(
+            node_strings, ["19+PR_32,673", "41,992", "1+PR_652,40+53", "4,5+6"]
+        )
+
+    def test_identify_concerted_rxns_via_intermediates(self):
 
+        v1 = loadfn(
+            os.path.join(test_dir, "identify_concerted_intermediate_list_v1.json")
+        )
         with open(
-            os.path.join(test_dir, "RN_unittest_reactions_list.json"), "rb"
-        ) as handle:
-            reactions = pickle.load(handle)
+            os.path.join(
+                test_dir, "identify_concerted_via_intermediate_unittest_RN.pkl"
+            ),
+            "rb",
+        ) as input:
+            RN_loaded = pickle.load(input)
+
+        v1_processed = []
+        for r_old in v1:
+            a = r_old[0]
+            b = r_old[1]
+            if None in b:
+                b.remove(None)
+            if len(a) == 2 and len(b) == 2:
+                inter = 0
+                inter = set(a).intersection(set(b))
+                for i in inter:
+                    a.remove(i)
+                    b.remove(i)
+            a.sort()
+            b.sort()
+            c = [a, b]
+            v1_processed.append(c)
+
+        v2_unique, v2_all = RN_loaded.identify_concerted_rxns_via_intermediates(
+            RN_loaded, single_elem_interm_ignore=[]
+        )
+        RN_loaded.add_concerted_rxns(RN_loaded, v2_unique)
+        v1_set = set(map(lambda x: repr(x), v1_processed))
+        v2_set = set(map(lambda x: repr(x), v2_unique))
+        inter = list(v1_set.intersection(v2_set))
+        v1_v2 = list(map(lambda y: literal_eval(y), v1_set - v2_set))
+        v2_v1 = list(map(lambda y: literal_eval(y), v2_set - v1_set))
+        dumpfn(v2_unique, "add_concerted_unittest_rxn_list.json")
```

Comment:
Do we want to be dumping this json every time the test is run? Maybe I'm misunderstanding, but I would have expected that we dump it once to generate the file to be tested against, then for every test in the future we load it and compare to it.

Reply:
Yes i totally forgot to delete that line. Will clean it up.